storage/innobase/mtr/mtr0mtr.cc

```diff
@@ -234,7 +234,13 @@ static void insert_imported(buf_block_t *block)
   if (block->page.oldest_modification() <= 1)
   {
     log_sys.latch.rd_lock(SRW_LOCK_CALL);
-    const lsn_t lsn= log_sys.last_checkpoint_lsn;
+    /* For unlogged mtrs (MTR_LOG_NO_REDO), we use the current system LSN. The
+    mtr that generated the LSN is either already committed or in mtr_t::commit.
+    Shared latch and relaxed atomics should be fine here as it is guaranteed
+    that both the current mtr and the mtr that generated the LSN would have
+    added the dirty pages to flush list before we access the minimum LSN during
+    checkpoint. */
+    const lsn_t lsn= log_sys.get_lsn();
```

Comment:
Thank you. This is the actual fix; the rest is `DEBUG_SYNC` instrumentation for the test. In retrospect, this corresponds to what was done in 89a5a8d234832ef9ed5ee814e4db42c636fcde1e to fix a similar bug in 10.5 and 10.6.

The commit might explicitly mention that `log_checkpoint_low()` is covered by an exclusive `log_sys.latch`, which would prevent any concurrent `mtr_t::commit()`.

Reply:
ok, done.